Python/Product/VSInterpreters/Interpreter/WorkspaceInterpreterFactoryProvider.cs

```diff
@@ -223,13 +225,32 @@ class WorkspaceInterpreterFactoryProvider : IPythonInterpreterFactoryProvider, I
             }
         }
 
-        private void OnFileCreated(object sender, FileSystemEventArgs e) {
+        private void OnFileCreatedDeletedRenamed(object sender, FileSystemEventArgs e) {
             lock (_factories) {
                 try {
-                    if (string.Compare(Path.GetFileName(e.FullPath), "python.exe", StringComparison.OrdinalIgnoreCase) == 0) {
-                        _pythonExecutableCreated = true;
+                    if (Directory.Exists(e.FullPath) && e.ChangeType == WatcherChangeTypes.Renamed) {
+                        RenamedEventArgs renamedFileInformation = e as RenamedEventArgs;
+                        if (renamedFileInformation != null) {
+                            string interpreterFactoryKey = _factories
+                                .Where(a => PathUtils.IsSameDirectory(a.Value.Configuration.PrefixPath, renamedFileInformation.OldFullPath))
+                                .Select(a => a.Key)
+                                .FirstOrDefault();
+
+                            if (!string.IsNullOrEmpty(interpreterFactoryKey)) {
+                                _refreshPythonInterpreters = true;
+                            }
+                        }
```

Comment:
Could be a bit simpler:
if (_factories.Values.Where(a => PathUtils.IsSameDirectory(a.Configuration.PrefixPath, renamedFileInformation.OldFullPath)).Any()) {
    _refreshPythonInterpreters = true;
}

Reply:
Sorry, there's actually even simplier:
```
if (_factories.Values.Any(a => PathUtils.IsSameDirectory(a.Configuration.PrefixPath, renamedFileInformation.OldFullPath))) {
_refreshPythonInterpreters = true;
}```

Reply:
Will be updated in next commit